src/ApiService/ApiService/OneFuzzTypes/Model.cs

```diff
@@ -471,19 +475,33 @@ public class NotificationTemplateConverter : JsonConverter<NotificationTemplate>
     public override NotificationTemplate? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) {
         using var templateJson = JsonDocument.ParseValue(ref reader);
         try {
-            return templateJson.Deserialize<AdoTemplate>(options);
-        } catch (JsonException) {
+            return ValidateDeserialization(templateJson.Deserialize<AdoTemplate>(options));
+        } catch (Exception ex) when (
+            ex is JsonException
+            || ex is ArgumentNullException
+            || ex is ArgumentOutOfRangeException
+        ) {
 
         }
 
         try {
-            return templateJson.Deserialize<TeamsTemplate>(options);
-        } catch (JsonException) {
+            return ValidateDeserialization(templateJson.Deserialize<TeamsTemplate>(options));
+        } catch (Exception ex) when (
+            ex is JsonException
+            || ex is ArgumentNullException
+            || ex is ArgumentOutOfRangeException
+        ) {
+
         }
 
         try {
-            return templateJson.Deserialize<GithubIssuesTemplate>(options);
-        } catch (JsonException) {
+            return ValidateDeserialization(templateJson.Deserialize<GithubIssuesTemplate>(options));
+        } catch (Exception ex) when (
+            ex is JsonException
+            || ex is ArgumentNullException
+            || ex is ArgumentOutOfRangeException
+        ) {
+
         }
         throw new JsonException("Unsupported notification template");
```

Comment:
so if github template throws JsonException, then we'll get here and throw "Unsupported notification template"

is that correct logic ?

Reply:
That's right, there's only those 3 templates that we support.